client/src/game/background.js

```diff
@@ -42,51 +56,166 @@ class Background {
   }
 
   drawNebulas () {
-    let generationChance = this.NEBULA_GENERATION[this.userSettings.map.nebulaDensity]
 
-    if (generationChance === 0) {
-      return
+    let NEBULA_FREQUENCY = this.userSettings.map.background.nebulaFrequency
+    let NEBULA_DENSITY = this.userSettings.map.background.nebulaDensity
+    let GENERATE_STARS = this.userSettings.map.background.backgroundStars == 'enabled'
+    let NEBULA_COLOR1 = parseInt(this.userSettings.map.background.nebulaColor1.replace(/^#/, '').substring(0, 6), 16)
+    let NEBULA_COLOR2 = parseInt(this.userSettings.map.background.nebulaColor2.replace(/^#/, '').substring(0, 6), 16)
+    let NEBULA_COLOR3 = parseInt(this.userSettings.map.background.nebulaColor3.replace(/^#/, '').substring(0, 6), 16)
```

Comment:
Can we trust the user isn't going to enter some stupid settings? Validation on the settings form or some kind of fallback if the setting cannot be extracted?

```
let NEBULA_COLOR1 = '#...'

try {
// Try to extract the setting
} catch {
// Can't do shit, use the default value.
}
```

Reply:
The above in a private function would be nice! `getNebulaSettingColour`?

Reply:
huum, i'll do some more validation and a fallback if the colors is outside range or there are random string giberish